src/dune_rules/dune_package.ml

```diff
@@ -466,8 +466,7 @@ module Or_meta = struct
            with
            | x -> Ok x
            | exception User_error.E message -> Error message
-           | exception exn ->
-             (* XXX stop catching code errors, invalid args, etc. *)
+           | exception ((Sys_error _ | End_of_file) as exn) ->
              Error (User_message.make [ Exn.pp exn ]))
```

Comment:
You can convert them to to proper error messages as well.

1. `Unexpected end of file when reading foo`
2. `[ Pp.textf "Failed to read %s." file; Pp.text sys_error ]`

Reply:
Good idea, fixed that in the latest push.